src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -1000,11 +1001,18 @@ def calculate_jvm_modulizable_targets(self):
         def is_jvm_or_resource_target(t):
             return isinstance(t, (JvmTarget, JvmApp, JarLibrary, Resources))
 
+        def is_synthetic_or_can_derive_synthetic(t):
+            return t.is_synthetic or isinstance(t, JavaThriftLibrary)
```

Comment:
Would it be safe to include `resources(..)` targets in this list as well?

Reply:
discussed offline. resources would not need to be filtered out here.